### PR TITLE
Add lexing/parsing for magic vars

### PR DESCRIPTION
**Pull request by nobodywasishere**

### CODE REVIEW

File: spec/parser/magic_spec.cr [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+
+      a = __LINE__
+      EXPR
+
+      node.should be_a LC::Assign
+      node = node.as(LC::Assign)
+
+      node.value.should be_a LC::IntLiteral
+      node.value.as(LC::IntLiteral).value.should eq 2
```
Comment (devnote-dev) [inline spec/parser/magic_spec.cr:16]: Shouldn't this be 3? I don't believe the line count starts from 0.

**Review by devnote-dev — CHANGES_REQUESTED** [review]

File: spec/parser/magic_spec.cr (hunk 1 above)
Comment (nobodywasishere) [reply spec/parser/magic_spec.cr:16]: The expr being parsed is `"\n\na = __LINE__"`. IIRC heredocs line 0 is the first line after the opener

Comment (devnote-dev) [reply spec/parser/magic_spec.cr:16]: Right, but the heredoc isn't part of the parsed content: https://carc.in/#/r/h150

Comment (nobodywasishere) [reply spec/parser/magic_spec.cr:16]: Ahh I understand now, my bad.

**Review by devnote-dev — APPROVED** [review]

**Pull request merged**